examples/cpp/dds/BasicConfigurationExample/BasicConfiguration_main.cpp

```diff
@@ -60,145 +60,166 @@ int main(
     int domain = 0;
     bool async = false;
     TransportType transport = DEFAULT;
+    int hops = -1;
     bool reliable = false;
     bool transient = false; // transient local
-    if (argc > 1)
+                            //
+    argc -= (argc > 0);
+    argv += (argc > 0); // skip program name argv[0] if present
+    option::Stats stats(true, usage, argc, argv);
+    std::vector<option::Option> options(stats.options_max);
+    std::vector<option::Option> buffer(stats.buffer_max);
+    option::Parser parse(true, usage, argc, argv, &options[0], &buffer[0]);
+
+    if (parse.error())
     {
-        if (!strcmp(argv[1], "publisher"))
-        {
-            type = PUBLISHER;
-        }
-        else if (!strcmp(argv[1], "subscriber"))
-        {
-            type = SUBSCRIBER;
-        }
-        // check if first argument is help, needed because we skip it when parsing
-        else if (!(strcmp(argv[1], "-h") && strcmp(argv[1], "--help")))
-        {
-            option::printUsage(fwrite, stdout, usage, columns);
-            return 0;
-        }
-        else
+        option::printUsage(fwrite, stdout, usage, columns);
+        return 1;
+    }
+
+    if (options[optionIndex::HELP])
+    {
+        option::printUsage(fwrite, stdout, usage, columns);
+        return 0;
+    }
+
+    // Decide between publisher or subscriber
+    try
+    {
+        if (parse.nonOptionsCount() != 1)
         {
-            std::cerr << "ERROR: first argument can only be <publisher|subscriber>" << std::endl;
-            option::printUsage(fwrite, stdout, usage, columns);
-            return 1;
+            throw 1;
```

Comment:
NIT: print a different trace for this error?

Reply:
or maybe we can rephrase it to be valid on all cases